pylint/lint/base_options.py

```diff
@@ -391,6 +391,15 @@ def _make_linter_options(linter: PyLinter) -> Options:
                 "positives when analysed.",
             },
         ),
+        (
+            "no-header",
```

Comment:
Sorry I did not think of it earlier but let's avoid the double negative.
```suggestion
            "display-header",
```

Reply:
I personally think that's not great from a user perspective.

But I'll implement 🤷‍♀️

Reply:
Well then we'd need to handle the error case when the formatter does not have header, or the option is useless because we're just listing all the message, etc. It seems like a lot of work. What do you think @DanielNoord ?

Reply:
I still think a reporter makes more sense. We don't have reporter specific options as they aren't `ArgumentProviders`. You only set this once in your config file so it doesn't seem to problematic to me.

I'd rather have 20 different reporters that all do something different than having to think of 20 different options for one reporter and how they interact with each other. If we're missing a combination of options we can just make a 21st reporter, while if we have one reporter we can't change anything without breaking some workflow for some user somewhere.